cvmfs/network/jobinfo.h

```diff
@@ -33,106 +33,197 @@ namespace download {
 /**
  * Contains all the information to specify a download job.
  */
-struct JobInfo {
-  const std::string *url;
-  bool compressed;
-  bool probe_hosts;
-  bool head_request;
-  bool follow_redirects;
-  bool force_nocache;
-  pid_t pid;
-  uid_t uid;
-  gid_t gid;
-  void *cred_data;  // Per-transfer credential data
-  InterruptCue *interrupt_cue;
-  cvmfs::Sink *sink;
-  const shash::Any *expected_hash;
-  const std::string *extra_info;
+class JobInfo {
+ private:
+  /// Pipe used for the return value
+  UniquePtr<Pipe<kPipeDownloadJobsResults> > pipe_job_results;
+  const std::string *url_;
+  bool compressed_;
+  bool probe_hosts_;
+  bool head_request_;
+  bool follow_redirects_;
+  bool force_nocache_;
+  pid_t pid_;
+  uid_t uid_;
+  gid_t gid_;
+  void *cred_data_;  // Per-transfer credential data
+  InterruptCue *interrupt_cue_;
+  cvmfs::Sink *sink_;
+  const shash::Any *expected_hash_;
+  const std::string *extra_info_;
 
   // Allow byte ranges to be specified.
-  off_t range_offset;
-  off_t range_size;
-
-  /**
-   * Tells whether the error is because of a non-existing file. Should only
-   * be called if error_code is not kFailOk
-   */
-  bool IsFileNotFound();
+  off_t range_offset_;
+  off_t range_size_;
 
   // Internal state, don't touch
```

Comment:
It's already private
```suggestion
  // Internal state
```